crates/uv-pep508/src/marker/parse.rs

```diff
@@ -239,12 +242,23 @@ pub(crate) fn parse_marker_key_op_value<T: Pep508Url>(
                 value,
             })
         }
+        // `extras in "test"` or `dependency_groups not in "dev"` are invalid.
+        MarkerValue::MarkerEnvList(key) => {
```

Comment:
This is a bit more strict than before, since there is no legacy with those markers.